Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -2100,7 +2100,9 @@ void Document::update_readiness(HTML::DocumentReadyState readiness_value)
 
     if (readiness_value == HTML::DocumentReadyState::Complete) {
         if (navigable() && navigable()->is_traversable()) {
-            HTML::HTMLLinkElement::load_fallback_favicon_if_needed(*this).release_value_but_fixme_should_propagate_errors();
+            if (is<HTML::HTMLHtmlElement>(this->document_element())) {
```

Comment:
According to the spec on how to load a favicon, here https://html.spec.whatwg.org/multipage/links.html#rel-icon:the-link-element-3, we can skip trying to load a favicon if the url of the document is not an http or https scheme.

The way we create the DOM::Document for SVG documents is a bit hard to track down, but it actually happens in SVG/SVGDecodedImageData.cpp, in the create() method. that's where we call `DOM::Document::create_and_initialize()` with the navigation params provided just above. 

Those params include a fetch response with the url of the parent page.

It's possible that putting in a dummy url there that's not an http{s} scheme would also work around the issue? This is all really an artifact of how we go about creating these documents.

@trflynn89 am I barking up the wrong tree? or should we go with this hack to fix our hacks :thinking:

Reply:
I'm not sure that we shouldn't be loading fallback favicons for SVG documents.

In Firefox, for example, if I navigate directly to https://cdn.shopify.com/b/shopify-brochure2-assets/1d6091ce14b313a3c9a19f237df898bb.svgz, they also try to load a fallback favicon:
<img width="872" alt="Screenshot 2024-03-05 at 1 19 41 PM" src="https://github.com/SerenityOS/serenity/assets/5600524/a647d73b-0239-4ff6-9a96-96f1ce0d1f4d">

Perhaps a better fix is to only load fallback icons for top-level documents?

Reply:
ah, in that case the check might be `navigable->is_top_level_traversable()` ? Except that we *do* create a top level traversable for every single SVGDecodedImageData as well, so that check might not actually do anything :)

Reply:
It is mainly so we can avoid loading a favicon for img links to an SVG, but apparently like you mentioned if won't work if you navigate directly to an SVG. For top-level documents do you mean something like this? :
```
bool Document::is_top_level_document()
{
    return (parent() == nullptr);
}
```
If this is what you meant then an image with an SVG URL inside a body element is a top-level document and so is an SVG loaded directly from a link in the navigation bar.

Reply:
> ah, in that case the check might be `navigable->is_top_level_traversable()` ? Except that we _do_ create a top level traversable for every single SVGDecodedImageData as well, so that check might not actually do anything :)

Yea it does not help

Reply:
I'm not particularly familiar with navigables yet - @ADKaster do we not have a way to ask "is this document the root document for the outermost frame?"

Reply:
I think in general that's what we want anyways, for HTML docs as well. i.e. we don't want to be fetching fallback favicons for iframes.

Reply:
Yes, that's what the top level traversable is supposed to be. But we abuse the document creation methods in a horribly hacky way in order to not crash the Parser when parsing SVGs

Reply:
We could tell the SVG code to create a child navigable, but that would cause other problems as those SVG documents aren't supposed to be *real* child Navigables for spec purposes. At least... I don't think they are. @kalenikaliaksandr ?

Reply:
why do we need Page and Navigable for each decoded SVG though? we need a Document to represent decoded SVG but wouldn't HTMLParser work without Page, Navigable and BrowsingContext?

Reply:
We want embedded `<img src=foo.svg>` files to render in a completely isolated context without access to the containing document, page, navigable or browsing context. That's why we make a fake page for them :)

Reply:
right. I think then we should mark navigables allocated for decoded SVGs, so we could skip them.